src/vsr/client.zig

```diff
@@ -388,12 +387,13 @@ pub fn Client(comptime StateMachine_: type, comptime MessageBus: type) type {
 
             // Eagerly release request message, to ensure that user's callback can submit a new
             // request.
-            self.unref(inflight.message);
-            inflight.message = undefined;
+            self.release(inflight.message);
+            assert(self.messages_available > 0);
 
-            // Even though we release our reference to the message, the user might have retained
+            // Even though we release our reference to the message, we might have retained
             // another one.
```

Comment:
Let's mention the `send` queue here; as written, the reader will wonder how _exactly_ do we retain the message.